solutions/noisg-19-feast.mdx

```diff
@@ -0,0 +1,83 @@
+---
+id: noisg-19-feast
+source: NOI
+title: 2019 - Feast
+author: Dong Liu
+---
+
+**Time Complexity:** $\mathcal{O}(N\log{\sum a[i]})$.
+
+<LanguageSection>
+
+<CPPSection>
+
+Like a normal Alien Trick problem, we're going to binary search on the maximum $\texttt{penalty}$.
+
+Now, for a given penalty, we're going to calculate the number of people in an optimal food arrangement such that we subtract $\texttt{penalty}$ for every person.
+
+Lets have $\texttt{dp}[i][j:\{0,1\}]$ represent the maximum sum of satisfaction of an arrangement of the first $i$ people, given that $j=0/1$ implies whether person $i$ is being used. Let $\texttt{cnt}[i][j]$ represent the number of people used in an optimal arrangement of $\texttt{dp}[i][j]$.  
```

Comment:
Also, don't you mean "plates" instead of "people"?

Reply:
oops